src/plugins/ccode/CMakeLists.txt

```diff
@@ -1,3 +1,12 @@
-include (LibAddMacros)
+if (DEPENDENCY_PHASE)
+	include (LibAddTest)
+	add_gtest (testmod_ccode
```

Comment:
why not add_plugin_test? Why in the dependency phase?

Reply:
> why not add_plugin_test?

I just copied this code from the [`type` plugin](https://github.com/ElektraInitiative/libelektra/blob/master/src/plugins/type/CMakeLists.txt). The [documentation](https://github.com/ElektraInitiative/libelektra/blob/master/cmake/Modules/LibAddPlugin.cmake#L250-L251) also states that you should use `add_gtest`. My favorite approach would be to just use `add_plugin`, but as far as I can tell this function and `add_plugin_test` [do not support C++ tests for a C++ plugin](https://github.com/ElektraInitiative/libelektra/blob/master/cmake/Modules/LibAddPlugin.cmake#L86).

> Why in the dependency phase?

Adding the plugin without `if (DEPENDENCY_PHASE)…` does not work:

```
CMake Error at cmake/Modules/LibAddTest.cmake:28 (add_executable):
  add_executable cannot create target "testmod_ccode" because another target
  with the same name already exists.  The existing target is an executable
  created in source directory "/Users/rene/Documents/Studium/Master
  Thesis/Configuration Parsing/Source/Elektra/src/plugins".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  src/plugins/ccode/CMakeLists.txt:1 (add_gtest)


CMake Error at cmake/Modules/LibAddMacros.cmake:80 (target_link_libraries):
  Attempt to add link library "elektra-core" to target "testmod_ccode" which
  is not built in this directory.
Call Stack (most recent call first):
  cmake/Modules/LibAddTest.cmake:34 (target_link_elektra)
  src/plugins/ccode/CMakeLists.txt:1 (add_gtest)


CMake Error at cmake/Modules/LibAddTest.cmake:40 (target_link_libraries):
  Attempt to add link library "gtest_main" to target "testmod_ccode" which is
  not built in this directory.
Call Stack (most recent call first):
  src/plugins/ccode/CMakeLists.txt:1 (add_gtest)


CMake Error at cmake/Modules/LibAddTest.cmake:43 (target_link_libraries):
  Attempt to add link library "gtest" to target "testmod_ccode" which is not
  built in this directory.
Call Stack (most recent call first):
  src/plugins/ccode/CMakeLists.txt:1 (add_gtest)


CMake Error at cmake/Modules/LibAddTest.cmake:51 (install):
  install TARGETS given target "testmod_ccode" which does not exist in this
  directory.
Call Stack (most recent call first):
  src/plugins/ccode/CMakeLists.txt:1 (add_gtest)
```

.

Reply:
> add_gtest

Yes, seems like we miss a feature in add_plugin.

>  Adding the plugin without if (DEPENDENCY_PHASE)… does not work:

I mean that ADDTESTING_PHASE should be the correct phase.

Reply:
> I mean that ADDTESTING_PHASE should be the correct phase.

After I changed the phase `add_gtest` [did not find the source files for the test at configuration time](https://travis-ci.org/ElektraInitiative/libelektra/jobs/393646495#L5642) any more. I fixed this problem by adding the code

```cmake
foreach (file ${ARG_SOURCES} ${source}.cpp)
  if (EXISTS "${file}")
    list (APPEND SOURCES "${file}")
  else ()
    list (APPEND SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/${file}")
  endif ()
endforeach ()
```

to `add_gtest`. However now the build fails printing an error message about `kdb.hpp`

```
In file included from ../src/plugins/ccode/coder.cpp:12:
../src/plugins/ccode/coder.hpp:17:10: fatal error: 'keyset.hpp' file not found
#include <keyset.hpp>
         ^~~~~~~~~~~~
```

. For now I undid the changes to fix the problem.